tests/unit/test_storage.py

```diff
@@ -86,33 +86,75 @@ def test_get_ceph_versions_mismatch_unavailable(self):
     @mock.patch.object(storage_01ceph, "CEPH_INFO", {})
     def test_get_ceph_pg_imbalance(self):
         result = {'pgs-per-osd': {
-                   'osd.0': 295,
-                   'osd.3': 214,
-                   'osd.15': 49,
-                   'osd.16': 316,
-                   'osd.17': 370,
-                   'osd.34': 392,
-                   'osd.37': 406,
-                   'osd.56': 209,
-                   'osd.72': 206}}
+            "osd.0": 399,
+            "osd.4": 32,
+        }}
         storage_01ceph.get_ceph_pg_imbalance()
         self.assertEqual(storage_01ceph.CEPH_INFO, result)
 
+    @mock.patch.object(storage_01ceph.helpers, "get_ceph_osd_df_tree_json",
+                       lambda: "{}")
     @mock.patch.object(storage_01ceph.helpers, "get_ceph_osd_df_tree",
-                       lambda: [])
+                       lambda: "")
     @mock.patch.object(storage_01ceph, "CEPH_INFO", {})
     def test_get_ceph_pg_imbalance_unavailable(self):
         storage_01ceph.get_ceph_pg_imbalance()
         self.assertEqual(storage_01ceph.CEPH_INFO, {})
 
+    @mock.patch.object(storage_01ceph.helpers, "get_ceph_osd_df_tree_json",
+                       lambda: "{}")
+    @mock.patch.object(storage_01ceph.helpers, "get_ceph_osd_df_tree",
+                       lambda: """ID CLASS WEIGHT  REWEIGHT SIZE   USE     DATA    OMAP META  AVAIL   %USE  VAR   PGS TYPE NAME
+-1       0.02939        - 30 GiB 3.0 GiB 8.4 MiB  0 B 3 GiB  27 GiB 10.03 1.00    - root default
+-3       0.00980        - 10 GiB 1.0 GiB 2.8 MiB  0 B 1 GiB 9.0 GiB 10.03 1.00    -      host juju-1157aa-ceph-1
+ 0   hdd 0.00980  1.00000 10 GiB 1.0 GiB 2.8 MiB  0 B 1 GiB 9.0 GiB 10.03 1.00  267        osd.0
+-7       0.00980        - 10 GiB 1.0 GiB 2.8 MiB  0 B 1 GiB 9.0 GiB 10.03 1.00    -      host juju-1157aa-ceph-2
+ 2   hdd 0.00980  1.00000 10 GiB 1.0 GiB 2.8 MiB  0 B 1 GiB 9.0 GiB 10.03 1.00  269        osd.2
+-5       0.00980        - 10 GiB 1.0 GiB 2.8 MiB  0 B 1 GiB 9.0 GiB 10.03 1.00    -      host juju-1157aa-ceph-3
+ 1   hdd 0.00980  1.00000 10 GiB 1.0 GiB 2.8 MiB  0 B 1 GiB 9.0 GiB 10.03 1.00  268        osd.1
+                    TOTAL 30 GiB 3.0 GiB 8.4 MiB  0 B 3 GiB  27 GiB 10.03
```

Comment:
I would really prefer to stick to using actual files for this kind of thing i.e. https://github.com/dosaboy/hotsos/blob/master/tests/unit/fake_data_root/sos_commands/ceph/ceph_osd_df_tree. I assume what you want here is for both files to be equivalent but one is non-json and the other json. If thats the case just go ahead and update both files and fixup existing unit tests that rely on their output. That will greatly reduce the amount of code in tests and the need for these extreme lambda functions. You can then also remove the global change in tox.ini since i assume it wont be needed anymore.

Also with your mock you are returning a string from get_ceph_osd_df_tree() but in reality it returns a list of string so this is not a valid mock value.